pkg/networkserver/registry.go

```diff
@@ -45,116 +46,84 @@ func logRegistryRPCError(ctx context.Context, err error, msg string) {
+	paths, replaced := registry.MatchReplacedEndDeviceFields(paths, w.fields)
 	dev, ctx, err := w.registry.GetByEUI(ctx, joinEUI, devEUI, paths)
 	if err != nil || dev == nil {
 		return dev, ctx, err
 	}
-	for _, d := range deprecated {
+	for _, d := range replaced {
 		d.GetTransform(dev)
 	}
 	return dev, ctx, nil
 }
 
-func (w deprecatedDeviceFieldRegistryWrapper) GetByID(ctx context.Context, appID ttnpb.ApplicationIdentifiers, devID string, paths []string) (*ttnpb.EndDevice, context.Context, error) {
-	paths, deprecated := matchDeprecatedDeviceFields(paths, w.fields)
+func (w replacedEndDeviceFieldRegistryWrapper) GetByID(ctx context.Context, appID ttnpb.ApplicationIdentifiers, devID string, paths []string) (*ttnpb.EndDevice, context.Context, error) {
+	paths, replaced := registry.MatchReplacedEndDeviceFields(paths, w.fields)
 	dev, ctx, err := w.registry.GetByID(ctx, appID, devID, paths)
 	if err != nil || dev == nil {
 		return dev, ctx, err
 	}
-	for _, d := range deprecated {
+	for _, d := range replaced {
 		d.GetTransform(dev)
 	}
 	return dev, ctx, nil
 }
 
-func (w deprecatedDeviceFieldRegistryWrapper) RangeByAddr(ctx context.Context, devAddr types.DevAddr, paths []string, f func(context.Context, *ttnpb.EndDevice) bool) error {
-	paths, deprecated := matchDeprecatedDeviceFields(paths, w.fields)
+func (w replacedEndDeviceFieldRegistryWrapper) RangeByAddr(ctx context.Context, devAddr types.DevAddr, paths []string, f func(context.Context, *ttnpb.EndDevice) bool) error {
+	paths, replaced := registry.MatchReplacedEndDeviceFields(paths, w.fields)
 	return w.registry.RangeByAddr(ctx, devAddr, paths, func(ctx context.Context, dev *ttnpb.EndDevice) bool {
 		if dev != nil {
-			for _, d := range deprecated {
+			for _, d := range replaced {
 				d.GetTransform(dev)
 			}
 		}
 		return f(ctx, dev)
 	})
 }
 
-func (w deprecatedDeviceFieldRegistryWrapper) SetByID(ctx context.Context, appID ttnpb.ApplicationIdentifiers, devID string, paths []string, f func(context.Context, *ttnpb.EndDevice) (*ttnpb.EndDevice, []string, error)) (*ttnpb.EndDevice, context.Context, error) {
-	paths, deprecated := matchDeprecatedDeviceFields(paths, w.fields)
+func (w replacedEndDeviceFieldRegistryWrapper) SetByID(ctx context.Context, appID ttnpb.ApplicationIdentifiers, devID string, paths []string, f func(context.Context, *ttnpb.EndDevice) (*ttnpb.EndDevice, []string, error)) (*ttnpb.EndDevice, context.Context, error) {
+	paths, replaced := registry.MatchReplacedEndDeviceFields(paths, w.fields)
 	dev, ctx, err := w.registry.SetByID(ctx, appID, devID, paths, func(ctx context.Context, dev *ttnpb.EndDevice) (*ttnpb.EndDevice, []string, error) {
 		if dev != nil {
-			for _, d := range deprecated {
+			for _, d := range replaced {
 				d.GetTransform(dev)
 			}
 		}
 		dev, paths, err := f(ctx, dev)
 		if err != nil || dev == nil {
 			return dev, paths, err
 		}
-		for _, d := range deprecated {
+		for _, d := range replaced {
+			if ttnpb.HasAnyField(paths, d.Old) {
```

Comment:
This is already handled by `registry.MatchReplacedEndDeviceFields`, remove

Reply:
Same; `paths` is from `f()`